modules/mono/editor/GodotTools/GodotTools.Core/StringExtensions.cs

```diff
@@ -7,7 +7,7 @@ namespace GodotTools.Core
 {
     public static class StringExtensions
```

Comment:
The check for an empty string is still relevant though, as a later part of the script relies on `path[path.Length - 1]`, which would break if the string was empty. Though after looking at some of the native `Path` functions, it seems that exceptions are thrown if the path is null or empty, so perhaps those exceptions can simply be added to the `StringExtensions` functions?

Reply:
For now, I'll make that check return `string.Empty` to guarantee the return value isn't `null`

Reply:
I'm not sure I understand, the method `NormalizePath` has a non-nullable string parameter so callers shouldn't be passing null, and since this is not public API we can guarantee that will never happen because we are always the caller. Are you saying we sometimes use null with this method?

About changing the return in the check to `string.Empty`, I'm not sure why that makes a difference since you are returning so you never reach the later part of the script that uses `path[path.Length - 1]`.

Reply:
We don't use `null` with that method, the `IsNullOrEmpty` really only checks for the empty string in practice. As such, it could be trimmed down to just `if (path == string.Empty)`

The return of `string.Empty` would only really be needed in the freak scenario where `null` *was* passed, which would make it return an empty string as if it were simply passed an empty string. So it's for the sake of ensuring a valid return if it were to return early; passing that first check means it will always have a length greater than zero

Reply:
But is `string.Empty` ever passed? I feel like we always pass a non-empty string to that method but if we want to still check for this I would probably prefer to keep the `string.IsNullOrEmpty`.

Reply:
It doesn't seem like it. I suppose then that the check can be removed altogether, which has precedence with several of our other path-related functions lacking such a check. It'd probably be worth having explicit exception thrown in several of these functions, but that's outside the scope of this PR